service/storage_service.cc

```diff
@@ -2559,6 +2562,9 @@ future<> storage_service::removenode(locator::host_id host_id, std::list<locator
 
                 // Step 6: Finish
                 req.cmd = node_ops_cmd::removenode_done;
+                if (ss._feature_service.quarantined_hosts) {
+                    req.hosts_to_quarantine.emplace(host_id, db_clock::now());
```

Comment:
In the replace operation (in a later commit) you also call `update_quarantined_hosts` immediately.
Why not here?